providers/implementations/keymgmt/mlkem_kmgmt.c

```diff
@@ -0,0 +1,575 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/core_dispatch.h>
+#include <openssl/core_names.h>
+#include <openssl/params.h>
+#include <openssl/err.h>
+#include <openssl/proverr.h>
+#include <openssl/evp.h>
+#include <openssl/rand.h>
+#include <openssl/self_test.h>
+#include "internal/param_build_set.h"
+#include <openssl/param_build.h>
+#include "prov/mlkem.h"
+#include "prov/implementations.h"
+#include "prov/providercommon.h"
+#include "prov/provider_ctx.h"
+#include "prov/securitycheck.h"
+#include <assert.h>
+
+#define BUFSIZE 1000
+#if defined(NDEBUG) || defined(OPENSSL_NO_STDIO)
+static void debug_print(char *fmt, ...)
+{
+}
+#else
+static void debug_print(char *fmt, ...)
+{
+    char out[BUFSIZE];
+    va_list argptr;
+
+    va_start(argptr, fmt);
+    vsnprintf(out, BUFSIZE, fmt, argptr);
+    va_end(argptr);
+    if (getenv("TEMPLATEKM"))
+        fprintf(stderr, "TEMPLATE_KM: %s", out);
+}
+#endif
+
+static void print_hex(const uint8_t *data, int len, const char *msg)
+{
+#ifndef NDEBUG
+    int i;
+
+    if (msg)
+        printf("%s: \n", msg);
+    for (i = 0; i < len; i++) {
+        if (i % 32 == 0)
+            printf("\n");
+        printf("%02x ", data[i]);
```

Comment:
BIO_dump_fp?

Reply:
used in https://github.com/openssl/openssl/commit/467e70f9af6925357f8f418f4e47e354c0a9fa15